packages/fx-core/src/component/generator/generator.ts

```diff
@@ -153,12 +156,25 @@ async function sampleDefaultOnActionError(
   context: GeneratorContext,
   error: Error
 ): Promise<void> {
+  await context.logProvider.error(error.message);
   switch (action.name) {
     case GeneratorActionName.FetchZipFromUrl:
-      throw new FetchZipFromUrlError(context.zipUrl!, error);
+      throw new FetchZipFromUrlError(context.zipUrl!);
     case GeneratorActionName.Unzip:
       throw new UnzipError();
     default:
       throw new Error(error.message);
   }
 }
+
+async function wrapError(run: () => Promise<void>) {
+  try {
+    await run();
+  } catch (error) {
+    if (error instanceof BaseComponentInnerError) {
+      throw error.toFxError();
```

Comment:
I think there are too many re-throws. Is it possible to transform the error in context.onActionError?
Like: 
```
throw (new FetchZipFromUrlError(context.zipUrl!)).toFxError();
```